src/Workspaces/Core/Portable/Utilities/NullableHelpers/NullableAttributes.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// This was copied from https://github.com/dotnet/coreclr/blob/60f1e6265bd1039f023a82e0643b524d6aaf7845/src/System.Private.CoreLib/shared/System/Diagnostics/CodeAnalysis/NullableAttributes.cs
+// and updated to have the scope of the attributes be internal.
```

Comment:
Are there associated analyzers that come with these attributes, or something else? Could you add the _why_ they are useful?

Reply:
These attributes are used by the compiler to understand "extra" nullability behaviors so it can give more accurate diagnostics.